src/Engine.sol

```diff
@@ -662,31 +654,33 @@ contract Engine is IEngine, EIP712, EIP7412, ERC2771Context {
     }
 
     /// @inheritdoc IEngine
-    function isPriceAbove(
-        bytes32 _assetId,
-        int64 _price,
-        uint64 _confidenceInterval
-    ) public view override returns (bool) {
-        PythStructs.Price memory priceData = ORACLE.getPrice(_assetId);
+    function isPriceAbove(uint128 _marketId, uint256 _price)
+        public
+        view
+        override
+        returns (bool)
+    {
+        (, uint256 fillPrice) = PERPS_MARKET_PROXY.computeOrderFees({
+            marketId: _marketId,
+            sizeDelta: 0
+        });
 
-        /// @dev although counterintuitive, a smaller confidence interval is more accurate.
-        /// The Engine must ensure the current confidence interval is not
-        /// greater (i.e. less accurate) than the confidence interval defined by the condition.
-        return priceData.price > _price && priceData.conf <= _confidenceInterval;
+        return fillPrice > _price;
     }
 
     /// @inheritdoc IEngine
-    function isPriceBelow(
-        bytes32 _assetId,
-        int64 _price,
-        uint64 _confidenceInterval
-    ) public view override returns (bool) {
-        PythStructs.Price memory priceData = ORACLE.getPrice(_assetId);
+    function isPriceBelow(uint128 _marketId, uint256 _price)
+        public
+        view
+        override
+        returns (bool)
+    {
+        (, uint256 fillPrice) = PERPS_MARKET_PROXY.computeOrderFees({
```

Comment:
**Thought:** My main concern with this, is that on the synthetix side, there is no staleness check for the value. I have gone through their code/logic, and assuming the oracle is of type `PYTH`, the following flow is used:
```
PerpsPrice.getCurrentPrice => NodeModule.processWithRuntime => NodeDefinition.process => PythNode.process
```
The following return flow occurs:
```
PythNode.process => (price, timestamp)
NodeDefinition.process  => (price, timestamp)
NodeModule.processWithRuntime => (price, timestamp)
PerpsPrice.getCurrentPrice => price
```
And under the hood of `PythNode.process`, it is using either `pyth.getEmaPriceUnsafe` or `pyth.getPriceUnsafe`.
As in the Pyth documentation, these methods say:
```solidity
    /// Users of this function should check the `publishTime` in the price to ensure that the returned price is
    /// sufficiently recent for their application. If you are considering using this function, it may be
    /// safer / easier to use either `getPrice` or `getPriceNoOlderThan`.
```
And:
```solidity
    /// Users of this function should check the `publishTime` in the price to ensure that the returned price is
    /// sufficiently recent for their application. If you are considering using this function, it may be
    /// safer / easier to use either `getEmaPrice` or `getEmaPriceNoOlderThan`.
 ```
This would be okay if synthetix was doing any staleness checks under the hood when executing `computeOrderFees`, but they don't.
Whereas previously you were using `pyth.getPrice`, which states in the Pyth docs:
```solidity
    /// @dev Reverts if the price has not been updated within the last `getValidTimePeriod()` seconds.
```

Hence there is a change in behaviour here that may not be intended - previously, the function would revert if the pyth oracle feed was stale. However using the `computeOrderFees` function, this check is not being done. It will just simply return a `fillPrice` and use that `fillPrice` regardless of how out of date it is.

This could lead to unintended behaviour - whereas previously this check would fail if the oracle was out of date, now it would pass - leaving this check to do be done on the synthetix side, which should then fail when calling `settleOrder`.

Reply:
It seems to me that `AsyncOrderModule.commitOrder()` follows the same path for defining the fill price. How then, can calls to `AsyncOrderModule.commitOrder()` revert if price feeds are stale, but not calls to `AsyncOrderModule.computeOrderFees()`?

Maybe I should share Troy's insights as well:

> I think you're correct that you should focus on the actual fill price instead of the raw pyth price, otherwise people could feel rugged. One example would be if a big liquidation goes through, the user could maybe catch the premium/discount if their order is submitted before the arb. But even better than the fill price returned by the contract, the way the current V3 contracts work they make sure that price is pretty fresh, or the prompt the user to go fetch more data. This sort of marries the "onchain" with the "offchain" stuff.

and

> Another complication is that perps could have MORE settlement strategies which implement new pricing and oracles. If you hard-code one oracle in here, I think it would be bad news in the future.

and 

> you can use confidence but would probably want to pay attention to the side of the trade. If the entire confidence interval was "below" then I don't think there is harm in filling that trade

Reply:
Final verdict:

These changes are acceptable because there are two contexts in which Conditional Orders are executed.

1. When the Conditional Order requires on-chain verification (see `requireVerified == true`), then at the moment of calling `Engine.execute()` the conditions *must* be met, and only after a successful on-chain verification will submitting an async order occur. The proposed change to check price may return false positive results when the price is stale during the instance of condition verification; however, in this specific context, calling `IPerpsMarketProxy.commitOrder()` will **fail** because the price is stale.

2. When the Conditional Order *does not* require on-chain verification (see `requireVerified == false`), then at the moment of calling `Engine.execute()`, the conditions member variable is **irrelevant**. The conditions are only relevant when on-chain verification is required when executing a Conditional Order.

Noteworthy mention:
In the case of a stale price, `canExecute` may return false positive results. However, at the time of async order submission, the order will fail if the price is stale and conditions must be verified on-chain. The same cannot be said for direct Oracle queries. If the Oracle query returns a false positive result, the async order submission could be successful. This introduces a higher degree of uncertainty. 

Thus, the question remains: which is a better source of truth? The order is executed against the Synthetix system. Relying on a query of this system for a price condition seems suitable, especially given Synthetix's nature of changing things abruptly and without warning.

Reply:
Based on our conversations, I have come to the conclusion that this change is not dangerous, as long as there is sufficient documentation explaining the potential limitations, and the front-end are made aware of these.